syz-fuzzer/machine_info.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2020 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"runtime"
+	"strings"
+)
+
+func CollectMachineInfo() ([]byte, error) {
+	if runtime.GOOS != "linux" {
+		return []byte{}, nil
+	}
+
+	type machineInfoFunc struct {
+		name string
+		fn   func() ([]byte, error)
+	}
+
+	allMachineInfo := []machineInfoFunc{
+		{"CPU Info", readCPUInfo},
+		{"KVM", readKVMInfo},
+	}
+
+	var buffer bytes.Buffer
+
+	for _, pair := range allMachineInfo {
+		buffer.WriteString(fmt.Sprintf("[%s]\n", pair.name))
```

Comment:
bytes.Buffer implements io.Writer, simpler (and more efficient) way to express this is:
``
fmt.Fprintf(buf, "[%s]\n", pair.name)
```